### ISSUE TITLE
Interface improvement: consistent return format between `swt` and `swt2`

### ISSUE THREAD

**Issue by ThomasA** [PyWavelets/pywt#81]
`swt` returns a list of coefficient tuples in descending order of wavelet level, while `swt2` returns its list of coefficient tuples in ascending order. This seems confusing and could easily lead to programming mistakes when for users of both functions.

I realise that changing the interface may be dangerous as well for reasons of backward compatibility.

**Comment by rgommers** [comment]
There's also the asymmetry in the API: ``swt2`` has a positional argument ``level`` why for ``swt`` it's optional (``level=None``).

And that one is in Python and one is in Cython is weird.

**Comment by rgommers** [comment]
Regarding the order, that's indeed confusing. It's a pretty bad backwards compat break, but I suspect that we'll want to make it. There are quite a few places where we'd like to break backwards compat in order to improve the API, and doing that sooner rather than later would be good.

**Comment by rgommers** [comment]
Unfortunately there's no good way to make a gradual change with a deprecation warning here.

**Comment by ThomasA** [comment]
How about retaining the existing version for a while, letting all calls 
of the function raise a deprecation warning that the order of the 
coefficients will change in a future version? That's certainly not 
optimal, but at least it gives regular users a chance to notice what's 
coming.

**Comment by rgommers** [comment]
One release with a ``FutureWarning`` might indeed make sense. But we'll still have to make the change at some point, and the longer we wait the more users we accumulate (if things go as planned....).

**Comment by rgommers** [comment]
It may be worth checking if this isn't the result of trying to match the Matlab output.

**Comment by ThomasA** [comment]
+1

**Comment by ThomasA** [comment]
It seems the corresponding functions in Matlab only handle one level at a time, so this ambiguity is not possible in Matlab:

    >> help swt
    swt - Discrete stationary wavelet transform 1-D

    This MATLAB function computes the stationary wavelet decomposition of the signal
    X at level N, using 'wname'.

    SWC = swt(X,N,'wname')
    SWC = swt(X,N,Lo_D,Hi_D)

    See also dwt, wavedec

    >> help swt2
    swt2 - Discrete stationary wavelet transform 2-D

    This MATLAB function or [A,H,V,D] = swt2(X,N,'wname') compute the stationary
    wavelet decomposition of the matrix X at level N, using 'wname'.

    SWC = swt2(X,N,'wname')
    [A,H,V,D] = swt2(X,N,'wname')
    SWC = swt2(X,N,Lo_D,Hi_D)
    [A,H,V,D] = swt2(X,N,Lo_D,Hi_D)

    See also dwt2, iswt2, wavedec2

Assuming pywt's functions are not supposed to mimick some *other* wavelet-related functions in Matlab, it seems there is no problem in relation to Matlab in changing the interfaces.

**Comment by ThomasA** [comment]
It seems swt2 is the unconventional function here. wavedec and wavedec2 are consistent with each other and return coefficients in the same order as swt.

**Comment by ThomasA** [comment]
level=None in swt is indeed weird. It obscures the fact that it actually defaults to the maximum possible number of levels further down in _swt while swt2 has no default value.

**Comment by grlee77** [comment]
@rgommers 
Re:  "And that one is in Python and one is in Cython is weird"
``swt2`` just calls ``swt`` repeatedly via a python loop the same way that ``dwt2`` calls ``dwt`` repeatedly via a python loop.  I moved some loops out of the python code and into the cython side for ``dwtn`` and ``idwtn`` in #68.  I think a similar refactoring (likely just moving entirely into the cython side) could be done for ``swt2``/``iswt2`` and ``dwt2``/``idwt2`` here as a separate PR.  Using the #68 PR code, it is approximately 4 times faster on my machine to do a 512x512 transform with ``dwtn`` than ``dwt2``.

**Comment by grlee77** [comment]
@ThomasA
I agree that swt2 is the one that would need to have the ordering changed to be consistent.  (i.e. after the change "coarser" resolutions should precede the "finer" ones in the output as in the wavedec2 case).  At least, I am certain that is the usual convention for storing the coefficients for the DWT.  I am less familiar with the expected behaviour in the undecimated case.

It seems that ``swt2`` is a multilevel function analagous to ``wavedec2`` rather than the single-level ``dwt2``.  Thus ``swt2`` is an outlier in that it is in multidim.py, but provides functionality similar to the routines in multilevel.py.

**Comment by rgommers** [comment]
I think we should rename all Python files so they start with an underscore (i.e. ``_multidim, _multilevel``). Then we're also are free to move ``swt2`` to a more logical place.

**Comment by ThomasA** [comment]
Are you working on refactoring iswt2? You should be aware that the currently proposed implementation of iswt2 does not work correctly. I am working on an alternative implementation in which I base it on repeatedly calling idwt2 similar to the way the currently proposed iswt calls idwt.

**Comment by grlee77** [comment]
No, I am not working on anything related to swt2 or iswt2 currently.  I did open issue #83 to remind me to take a look at some point in the future to speeding up the 2D transforms.

In regards to the renaming proposed by @rgommers, I did rename multilevel->_multilevel in #67 as it was suggested there.  However, now I see that @aaren has already implemented all of the proposed underscored name changes in #55:  https://github.com/rgommers/pywt/pull/55/files#diff-abe017763c973920f39d4d1d05d1abaeR16
